### PR TITLE
Player Character class

### PR DESCRIPTION
**Pull request by ZeroNeji**
This is the player character class that has the attributes from the Jira. I made the variables that I thought won't change much during the game into private. Methods are in PascalCase and variables are in camelCase. The constructor will deal with the first part of the character creation process, SetStats for the second, SetEquiped for the third, and SetSpells for the fourth. I have the attributeSaveProficiency as an int without a set so far because I wasn't sure what to do with it. The question for the discord was for this (question: did you mean for strSaveProf to be the (strength-10)/2 or a way to tell if they have strength save proficiency?). I made the getter in => format to shorten the file to make it easier to go through.

### CODE REVIEW

File: PlayerCharacter.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
+using System;
+using UnityEngine;
+
+class PlayerCharacter{
+  public string name;
+  public string className;
+  private string race;
+  private string subrace;
```
Comment (B-Rosario) [inline PlayerCharacter.cs]: I don't think subrace is necessary; we can just include the subrace in the race (for example High/Wood Elf or Variant Human).

File: PlayerCharacter.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  private int charisma;
+  private int level;
+  private int maxhp;
+  public int currenthp;
+  public int armorClass;
+  private int initiative;
+  private int speed;
+  public currentSpeed;
+
+  private int strSaveProficiency;
```
Comment (B-Rosario) [inline PlayerCharacter.cs]: Proficiencies should be Booleans.

File: PlayerCharacter.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  private int constitution;
+  private int intelligence;
+  private int charisma;
+  private int level;
+  private int maxhp;
+  public int currenthp;
+  public int armorClass;
+  private int initiative;
+  private int speed;
+  public currentSpeed;
```
Comment (B-Rosario) [inline PlayerCharacter.cs]: Data type?

File: PlayerCharacter.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  private int profiencyBonus;
+
+  private string[] armorProficencies;
+  private string[] weaponProficencies;
+
+  public string[] equipment;
+  public int[] spellSlots = new int[10];
+  private string[] spells;
+  
+  public string condition;
```
Comment (B-Rosario) [inline PlayerCharacter.cs]: Should also be an array.

File: PlayerCharacter.cs [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  public currentSpeed;
+
+  private int strSaveProficiency;
+  private int dexSaveProficiency;
+  private int conSaveProficiency;
+  private int intSaveProficiency;
+  private int wisSaveProficiency;
+  private int chaSaveProficiency;
+
+  private int profiencyBonus;
```
Comment (B-Rosario) [inline PlayerCharacter.cs]: misspelled but not a big deal; should correct for clarity.

**Review by B-Rosario — DISMISSED** [review]
Check errors

File: PlayerCharacter.cs (hunk 1 above)
Comment (ZeroNeji) [reply PlayerCharacter.cs]: I've taken out subrace.

File: PlayerCharacter.cs (hunk 2 above)
Comment (ZeroNeji) [reply PlayerCharacter.cs]: I made them into bool, kept proficiencyBonus as int.

File: PlayerCharacter.cs (hunk 3 above)
Comment (ZeroNeji) [reply PlayerCharacter.cs]: added datatype

File: PlayerCharacter.cs (hunk 4 above)
Comment (ZeroNeji) [reply PlayerCharacter.cs]: made into an array, also made immunities and resistances array

File: PlayerCharacter.cs (hunk 5 above)
Comment (ZeroNeji) [reply PlayerCharacter.cs]: fixed

**Comment by ZeroNeji** [comment]
I've added a batch of changes. Putting it here so that you guys can see the progress.

**Review by B-Rosario — APPROVED** [review]

**Review by JaredCas0 — APPROVED** [review]

**Pull request merged**